### PR TITLE
refactor(ant,pkg-get,pkgutil,rpm,ssh): adjust towards current API

### PR DESCRIPTION
**Pull request by scop**

### CODE REVIEW

File: completions/ant [hunk 1]
```diff
@@ -9,7 +9,7 @@ _ant_parse_targets()
     # parse buildfile for targets
     while read -rd '>' line; do
         if [[ $line =~ \<(target|extension-point)[[:space:]].*name=[\"\']([^\"\']+) ]]; then
-            targets+=" ${BASH_REMATCH[2]}"
+            ret+=" ${BASH_REMATCH[2]}"
```
Comment (akinomyoga) [inline completions/ant:12]: `ret` needs to be initialized in advance. Also, can we use arrays?

```suggestion
            ret+=("${BASH_REMATCH[2]}")
```

File: completions/ant [hunk 2]
```diff
@@ -9,7 +9,7 @@ _ant_parse_targets()
     # parse buildfile for targets
```
Comment (akinomyoga) [inline completions/ant:9]:
```suggestion
    # parse buildfile for targets
    ret=()
```

or if we still use a space-separated result in the scalar `ret`,

```suggestion
    # parse buildfile for targets
    ret=""
```

File: completions/ant [hunk 3]
```diff
@@ -87,12 +87,12 @@ _comp_cmd_ant()
...
 
-        local targets
+        local ret
 
         # fill targets
-        _ant_parse_targets "$buildfile"
+        _comp_cmd_ant__targets "$buildfile"
 
-        COMPREPLY=($(compgen -W '$targets' -- "$cur"))
+        COMPREPLY=($(compgen -W '$ret' -- "$cur"))
```
Comment (akinomyoga) [inline completions/ant:95]:
```suggestion
        COMPREPLY=($(compgen -W '"${ret[@]}"' -- "$cur"))
```

File: completions/java [hunk 4]
```diff
@@ -103,15 +96,15 @@ _comp_cmd_java__classes()
...
 
     # find which sourcepath to use
-    _java_find_sourcepath
+    _comp_cmd_java__find_sourcepath
 
     # convert package syntax to path syntax
     cur=${cur//.//}
     # parse each sourcepath element for packages
-    for i in ${sourcepath//:/ }; do
+    for i in ${ret//:/ }; do
```
Comment (akinomyoga) [inline completions/java:107]: These changes overlap with #888. I'll later check both changes.

Comment (scop) [reply completions/java:107]: Yeah, that one's marked as draft so I wasn't quite sure of its status. But we can drop the java changes from here if you like, and address them later/elsewhere.

File: completions/ant (hunk 1 above)
Comment (scop) [reply completions/ant]: Ah, sure, thanks!

File: completions/java (hunk 4 above)
Comment (scop) [reply completions/java]: Removed java from here, added pkgutil, rpm, and ssh instead.

File: completions/ant (hunk 2 above)
Comment (scop) [reply completions/ant:9]: Actually this function is recursive, so initialization needs to happen outside of it. Addressed in the latest commit.

Comment (akinomyoga) [reply completions/ant:9]: Ah, I see. Thank you.

**Review by akinomyoga — APPROVED** [review]

**Pull request merged**